game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java

```diff
@@ -0,0 +1,87 @@
+package com.jzy.game.engine.util;
+
+import com.jzy.game.engine.util.StringUtil;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StringUtilTest {
+
+  @Test
+  public void isNullOrEmptyFalse() {
+    Assert.assertEquals(false, StringUtil.isNullOrEmpty("      `a "));
+  }
+
+  @Test
+  public void isNullOrEmptyTrue() {
+    Assert.assertEquals(true, StringUtil.isNullOrEmpty(null));
+    Assert.assertEquals(true, StringUtil.isNullOrEmpty("      "));
+  }
+
+  @Test
+  public void isIntegerEmpty() {
+    Assert.assertEquals(false, StringUtil.isInteger(""));
+    Assert.assertEquals(false, StringUtil.isInteger(null));
+  }
+
+  @Test
+  public void isIntegerTrue() {
+    Assert.assertEquals(true, StringUtil.isInteger("-1"));
+    Assert.assertEquals(true, StringUtil.isInteger("1"));
+    Assert.assertEquals(true, StringUtil.isInteger("+1"));
+    Assert.assertEquals(true, StringUtil.isInteger("+"));
+    Assert.assertEquals(true, StringUtil.isInteger("-"));
+  }
+
+  @Test
+  public void isIntegerFalse() {
+    Assert.assertEquals(false, StringUtil.isInteger("--"));
+    Assert.assertEquals(false, StringUtil.isInteger("++"));
+    Assert.assertEquals(false, StringUtil.isInteger("a"));
+  }
+
+  @Test
+  public void isDoubleEmpty() {
+    Assert.assertEquals(false, StringUtil.isDouble(""));
+    Assert.assertEquals(false, StringUtil.isDouble(null));
+  }
+
+  @Test
+  public void isDoubleTrue() {
+    Assert.assertEquals(true, StringUtil.isDouble("-1"));
+    Assert.assertEquals(true, StringUtil.isDouble("-1."));
+    Assert.assertEquals(true, StringUtil.isDouble("1."));
+    Assert.assertEquals(true, StringUtil.isDouble("1."));
+    Assert.assertEquals(true, StringUtil.isDouble("+1"));
+    Assert.assertEquals(true, StringUtil.isDouble("+1."));
+    Assert.assertEquals(true, StringUtil.isDouble("+"));
+    Assert.assertEquals(true, StringUtil.isDouble("-"));
+    Assert.assertEquals(true, StringUtil.isDouble("+."));
+    Assert.assertEquals(true, StringUtil.isDouble("-."));
+  }
+
+  @Test
+  public void isDoubleFalse() {
+    Assert.assertEquals(false, StringUtil.isDouble("--"));
+    Assert.assertEquals(false, StringUtil.isDouble("++"));
+    Assert.assertEquals(false, StringUtil.isDouble("a"));
+  }
+
+  @Test
+  public void lowerFirstChar() {
+    Assert.assertEquals("a!!!!!!!", StringUtil.lowerFirstChar("A!!!!!!!"));
```

Comment:
Use test case that shows that the remaining characters are not capitalized (capital ! == ! so this is not shown)